---
string name
string type
uint32 error